src/incubator/TextField/useFieldState.ts

```diff
@@ -0,0 +1,90 @@
+import {useCallback, useState, useEffect, useMemo} from 'react';
+import {TextInputProps} from 'react-native';
+import _ from 'lodash';
+import validators from './validators';
+
+export type Validator = Function | keyof typeof validators;
+
+export interface FieldStateProps extends TextInputProps {
+  validateOnStart?: boolean;
+  validateOnChange?: boolean;
+  validateOnBlur?: boolean;
+  /**
+   * A single or multiple validator. Can be a string (required, email) or custom function.
+   */
+  validate?: Validator | Validator[];
+}
+
+export default function useFieldState({
+  validate,
+  validateOnBlur,
+  validateOnChange,
+  validateOnStart,
+  ...props
+}: FieldStateProps) {
+  const [value, setValue] = useState(props.value);
+  const [isFocused, setIsFocused] = useState(false);
```

Comment:
Do we have to have `isFocused`? Each time a `TextField` is focused\blurred it is re-rendered.

Reply:
Yes of course. 
isFocused affect the field UI and other functionality. 

For example:
The Field's label changes its color depending on the focus state
A validator can triggered when blurring the field and as a result show an validation error message.. 

It has to trigger a re-render.